src/utils/automation-helper.js

```diff
@@ -29,10 +29,337 @@ export const getAutomation = async (automationId) => {
   return null;
 };
 
-export const checkIfTriggered = async (automation) => automation.runMode === 'triggered';
+const getAutomationRun = async (automationId, runId) =>
+  fetch(`../api/v1/automations/${automationId}/runs/${runId}`).then((res) => res.json());
 
-export const getAutomationFromItem = async (itemId) => {
-  const response = await fetch(`../api/v1/items/${itemId}`);
-  const data = await response.json();
-  return getAutomation(data.resourceId);
+const checkMessage = (data, translator) => {
+  const defaultMessage = translator.get('Object.ActionButton.Automation.DefaultAutomationMsg');
+  if (Array.isArray(data)) {
+    return { message: data.join(' ').length > 0 ? data.join(' ') : defaultMessage };
+  }
+  if (Object.keys(data).includes('message')) {
+    return data;
+  }
+  return { message: defaultMessage };
+};
+
+const parseOutput = (data, translator) => {
+  if (typeof data !== 'undefined') {
+    try {
+      const message = JSON.parse(data);
+      return checkMessage(message, translator);
+    } catch {
+      if (typeof data === 'object') {
+        return checkMessage(data, translator);
+      }
+      return { message: data };
+    }
+  }
+  return { message: translator.get('Object.ActionButton.Automation.DefaultAutomationMsg') };
+};
+
+// eslint-disable-next-line no-promise-executor-return
+const sleep = (time) => new Promise((resolve) => setTimeout(resolve, time));
+
+const automationRunPolling = async (automationId, runId, translator) => {
```

Comment:
I would probably have used a function that is called in a timeout. The function would call itself when the automation run is not completed. In that way you would get a better overview of the different states that can be returned from the API. Also you do not need to await in a for-loop

Reply:
I changed this based on your suggestion - it is now much easier to read :) 

I currently set a timeout of 10 minutes. Do we have a default timeout for things like this in the client or should we stick with 10 minutes?